_targets.R

```diff
@@ -0,0 +1,150 @@
+#
+# you must execute the command 'earthengine authenticate' in a zsh terminal 
+# before initializing this workflow. See the repository README for complete 
+# dependencies and troubleshooting.
+
+# RUNNING {TARGETS}:
+#
+# run tar_make() to run the pipeline
+# run tar_visnetwork() to see visual summary
+
+
+# Set up python virtual environment ---------------------------------------
+
+if(!dir.exists('env')) {
+  source('data_acquisition/src/pySetup.R')
+} else {
+  use_condaenv(file.path(getwd(), 'env'))
+}
+
+
+# Source functions --------------------------------------------------------
+
+source('data_acquisition/src/general_functions.R')
+source_python('data_acquisition/src/gee_functions.py')
+
+# Define {targets} workflow -----------------------------------------------
+
+# Set target-specific options such as packages.
+tar_option_set(packages = "tidyverse")
+
+# target objects in workflow
+list(
+  tar_target(name = checkYmlMod,
+             command = return(file.info(yaml_file)$mtime),
+             cue = tar_cue('always')), 
+  
+  tar_target(name = readYmlMod,
+             command = tar_read(checkYmlMod),
+             cue = tar_cue('always')),
+  
+  # load, format, save yml as a csv
+  tar_target(name = ymlSave,
+             command = {
+               readYmlMod
+               formatYaml(yaml_file)
+               },
+             packages = c('yaml', 'readr')),
+  
+  # track the yml file
+  tar_file_read(name = yml,
+                command = {
+                  readYmlMod
+                  tar_read(ymlSave)
+                  },
+                read = read_csv(!!.x),
+                packages = 'readr'),
+  
+  # load, format, save locs as an updated csv
+  tar_target(name = locsSave,
+             command = {
+               readYmlMod
+               grabLocs(yml)
+               },
+             packages = 'readr'),
+
+  # track locs file
+  tar_file_read(name = locs,
+                command = {
+                  readYmlMod
+                  tar_read(locsSave)
+                  },
+                read = read_csv(!!.x),
+                packages = 'readr'),
+  
+  # use location to get polygons for NHD
+  tar_target(name = polySave,
+             command = {
+               readYmlMod
+               getNHD(locsSave, ymlSave)
+               },
+             packages = c('nhdplusTools', 'sf', 'readr')),
+  
+  # track polygons file
+  tar_file_read(name = polygon, # this will throw an error if the configure extent does not include polygon
+                command = {
+                  readYmlMod
+                  tar_read(polySave)
+                  },
+                read = read_sf(!!.x),
+                packages = 'sf',
+                error = 'null'),
+  
+  # use polygon to calculate centers
+  tar_target(name = centersSave,
+             command = {
+               readYmlMod
+               calcCenter(polySave, ymlSave)
+               },
+             packages =  c('sf', 'polylabelr')),
```

Comment:
I've been wondering if there's a good way to communicate which packages are needed and which the user will need to install. I didn't realize I needed the polylabelr package, for example. Maybe this is something we just include in READMEs or something.

Reply:
YES!

I do think we should make it customary to use a modified version of [CM/KW's package loader](https://github.com/rossyndicate/ROSSyndicate_best_practices/pull/7) at the top of every _targets.R file, where all libraries that are required are installed and/or updated automatically.

```
package_installer <- function(x) {
    if (x %in% installed.packages()) {
      install.packages(x)
      print(paste0('{', x ,'} package is installed.')
    } else {
      print(paste0('{', x ,'} package is installed.')
    }
}

# fill in with packages that need to be checked for install:
packages <- c('tidyverse',
              'sf')

walk(packages, package_installer)
```